src/entity/Living.php

```diff
@@ -659,6 +663,32 @@ protected function entityBaseTick(int $tickDiff = 1) : bool{
 		return $hasUpdate;
 	}
 
+	protected function move(float $dx, float $dy, float $dz) : void{
+		parent::move($dx, $dy, $dz);
+
+		$frostWalkerLevel = $this->armorInventory->getBoots()->getEnchantmentLevel(VanillaEnchantments::FROST_WALKER());
+		if($frostWalkerLevel > 0 && (abs($dx) > 0.0001 || abs($dz) > 0.0001)){
```

Comment:
You can't use `dx` and `dz` here. These parameters are the _desired_ movement distance, not the actual distance moved.

Reply:
What should I use instead?

Reply:
You would need to record the position before and after the parent::move() call.

A possible option could be to override `checkGroundState()`, but it wasn't really intended for this usage, so it might break in other ways.